staging/src/k8s.io/pod-security-admission/policy/check_seccomp.go

```diff
@@ -0,0 +1,334 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/pod-security-admission/api"
+)
+
+const (
+	annotationKeyPod             = "seccomp.security.alpha.kubernetes.io/pod"
+	annotationKeyContainerPrefix = "container.seccomp.security.alpha.kubernetes.io/"
+	missingRequiredValue         = "<missing required value>"
+)
+
+func init() {
+	addCheck(CheckSeccompBaseline)
+	addCheck(CheckSeccompRestricted)
+}
+
+func fieldValue(f *field.Path, val string) string {
+	return fmt.Sprintf("%s=%s", f.String(), val)
+}
+
+func fieldValueRequired(f *field.Path) string {
+	return fmt.Sprintf("%s=%s", f.String(), missingRequiredValue)
+}
+
+func CheckSeccompBaseline() Check {
+	return Check{
+		ID:    "seccomp",
+		Level: api.LevelBaseline,
+		Versions: []VersionedCheck{
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 0),
+				CheckPod:       seccomp_1_0_baseline,
+			},
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 19),
+				CheckPod:       seccomp_1_19_baseline,
+			},
+		},
+	}
+}
+
+func CheckSeccompRestricted() Check {
+	return Check{
+		ID:    "seccomp",
+		Level: api.LevelRestricted,
+		Versions: []VersionedCheck{
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 0),
+				CheckPod:       seccomp_1_0_restricted,
+			},
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 19),
+				CheckPod:       seccomp_1_19_restricted,
+			},
+		},
+	}
+}
+
+func validSeccomp(t corev1.SeccompProfileType) bool {
+	return t == corev1.SeccompProfileTypeLocalhost ||
+		t == corev1.SeccompProfileTypeRuntimeDefault
+}
+
+func validSeccompAnnotation(annotation string) bool {
+	return annotation == corev1.SeccompProfileRuntimeDefault ||
+		strings.HasPrefix(annotation, corev1.SeccompLocalhostProfileNamePrefix)
+}
+
+// baseline policy checks
+
+// seccomp_1_0_baseline checks baseline policy on seccomp alpha annotation
+func seccomp_1_0_baseline(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	forbidden := sets.NewString()
+	podAnnotationField := field.NewPath("metadata").Child("annotations", annotationKeyPod)
+
+	if podMetadata.Annotations != nil {
+		if val, ok := podMetadata.Annotations[annotationKeyPod]; ok {
+			if val == corev1.SeccompProfileNameUnconfined {
+				forbidden.Insert(fieldValue(podAnnotationField, val))
+			}
+		}
+	}
+
+	for _, c := range podSpec.Containers {
```

Comment:
if you're iterating over containers, use visitContainersWithPath to make sure you also check ephemeralContainers properly (and any future container types)